nano/node/active_transactions.cpp

```diff
@@ -56,33 +58,34 @@ void nano::active_transactions::confirm_frontiers (nano::transaction const & tra
 		prioritize_frontiers_for_confirmation (transaction_a, is_test_network ? std::chrono::milliseconds (50) : std::chrono::seconds (2));
 
 		size_t elections_count (0);
-		std::unique_lock<std::mutex> lock (mutex);
+		lk.lock ();
 		while (!priority_cementable_frontiers.empty () && !stopped && elections_count < max_elections)
 		{
-			auto cementable_account = *priority_cementable_frontiers.begin ();
-			priority_cementable_frontiers.erase (priority_cementable_frontiers.begin ());
-			lock.unlock ();
+			auto cementable_account = *priority_cementable_frontiers.get<1> ().begin ();
+			priority_cementable_frontiers.get<1> ().erase (priority_cementable_frontiers.get<1> ().begin ());
+			lk.unlock ();
 			nano::account_info info;
 			auto error = node.store.account_get (transaction_a, cementable_account.account, info);
 			release_assert (!error);
 
 			if (info.block_count > info.confirmation_height && !this->node.pending_confirmation_height.is_processing_block (info.head))
 			{
-				auto block (this->node.store.block_get (transaction_a, info.head));
-				if (!this->start (block))
+				auto block (node.store.block_get (transaction_a, info.head));
+				lk.lock ();
+				auto root_exists = roots.find (block->qualified_root ()) != roots.end ();
+				lk.unlock ();
+				if (!root_exists && !start (block))
```

Comment:
I don't think we should check root existence here because it's in start () / add () function

Reply:
ok removed